packages/integration-tests/index.js

```diff
@@ -159,15 +161,134 @@ async function testWorkflow(stackName, bucketName, workflowName, inputFile) {
   }
 }
 
+/**
+ * set process environment necessary for database transactions
+ *
+ * @param {string} stackName - Cloud formation stack name
+ * @param {string} bucketName - S3 internal bucket name
+ * @returns {*} undefined
+ */
+function setProcessEnvironment(stackName, bucketName) {
+  process.env.internal = bucketName;
+  process.env.stackName = stackName;
+  process.env.CollectionsTable = `${stackName}-CollectionsTable`;
+  process.env.ProvidersTable = `${stackName}-ProvidersTable`;
+}
+
+/**
+ * add collections to database
+ *
+ * @param {string} stackName - Cloud formation stack name
+ * @param {string} bucketName - S3 internal bucket name
+ * @param {string} dataDirectory - the directory of collection json files
+ * @returns {Promise.<integer>} number of collections added
+ */
+async function addCollections(stackName, bucketName, dataDirectory) {
+  setProcessEnvironment(stackName, bucketName);
+  const filenames = await fs.readdir(dataDirectory);
+  const collections = [];
+  filenames.forEach((filename) => {
+    const collection = JSON.parse(fs.readFileSync(`${dataDirectory}/${filename}`, 'utf8'));
+    collections.push(collection);
+  });
+
+  const concurrencyLimit = process.env.CONCURRENCY || 3;
+  const limit = pLimit(concurrencyLimit);
+  const promises = collections.map((collection) => limit(() => {
+    const c = new Collection();
+    console.log(`adding collection ${collection.name}___${collection.version}`);
+    return c.delete({ name: collection.name, version: collection.version })
+      .then(() => c.create(collection));
+  }));
+  return Promise.all(promises).then((cs) => cs.length);
+}
+
+/**
+ * add providers to database
+ *
+ * @param {string} stackName - Cloud formation stack name
+ * @param {string} bucketName - S3 internal bucket name
+ * @param {string} dataDirectory - the directory of provider json files
+ * @returns {Promise.<integer>} number of providers added
+ */
+async function addProviders(stackName, bucketName, dataDirectory) {
+  setProcessEnvironment(stackName, bucketName);
```

Comment:
Lines 215-221 look pretty much exactly the same as 187-193. I wonder if some refactoring could be done to extract this common code out into a function.